File: 404.左叶子之和.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(!root)
            return 0;
        if(isLeaf(root->left))
            return root->left->val + sumOfLeftLeaves(root->right);
        return sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right);
    }

    bool isLeaf(TreeNode* root){
        if(!root)
            return false;
        if(!root->left && !root->right)
            return true;
        return false;
    }
};

File: 树/404.左叶子之和.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */


 var sumOfLeftLeaves = function (root) {
    let sum = 0;

    const dfs = (root) => {
        if (root === null)
            return;
        if (isLeaf(root.left))
            sum += root.left.val;
        dfs(root.left);
        dfs(root.right);
    }

    dfs(root);
    return sum;
};

const isLeaf = (root) => {
    if (root === null)
        return false;
    if (root.left === null && root.right === null) {
        return true;
    }
    return false;
}